AK/HashMap.h

```diff
@@ -90,6 +90,18 @@ class HashMap {
         return m_table.find(hash, predicate);
     }
 
+    // FIXME: Use some sort of Traits to get the comparison operation
+    template<typename Key>
+    [[nodiscard]] IteratorType find(Key const& value) requires(!IsTriviallyConstructible<K, Key> && requires { declval<Key>() == declval<K>(); })
+    {
+        return m_table.find(Traits<Key>::hash(value), [&](Entry const& entry) { return value == entry.key; });
+    }
+    template<typename Key>
+    [[nodiscard]] ConstIteratorType find(Key const& value) const requires(!IsTriviallyConstructible<K, Key> && requires { declval<Key>() == declval<K>(); })
```

Comment:
This is to prevent anything not being casted up to a u64 when required,
Although I should do that better with just explicit exclusion